include/picongpu/plugins/transitionRadiation/Particle.hpp

```diff
@@ -0,0 +1,224 @@
+/* Copyright 2017-2019 Finn-Ole Carstens
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+namespace picongpu
+{
+namespace plugins
+{
+namespace transitionRadiation
+{
+    /** Particle class for transition radiation calculation.
+     * 
+     * @param locationSet global position of the macro-particle
+     * @param momentumSet momentum of macro-particle
+     * @param charge
+     */
+    class Particle
+    {
+    private:
+        float3_X const & momentum;
+        float3_X const & location;
+        float_X const mass;
+
+    public:
+        HDINLINE 
+        Particle( 
+            float3_X const & locationSet, 
+            float3_X const & momentumSet,
+            float_X const massSet
+        ) :
+            location( locationSet ),
+            momentum( momentumSet ),
+            mass( massSet )
+        { 
+        }
+
+        //! @return momentum
+        HDINLINE 
+        float3_X 
+        getMomentum( ) const
+        {
+            return momentum;
+        }
+
+        //! @return normalized momentum
+        HDINLINE 
+        float_X 
+        getU( ) const
+        {
+            return calcU( getMomentum( ) );
+        }
+
+        //! @return velocity v = beta * c
+        HDINLINE
+        float_X 
+        getVel( ) const
+        {
+            return calcBeta( getMomentum( ) ) * picongpu::SPEED_OF_LIGHT;
+        }
+
+        //! @return polar angle phi of momentum
+        HDINLINE
+        float_X 
+        getMomPhi( ) const
+        {
+            return calcMomPhi( );
```

Comment:
You are hiding the `calc...` methods behind `get...` methods - that's Okay for me, but adds unnecessary code and might add confusion.  
@psychocoderHPC and @sbastrakov what do you think?

Reply:
I just had an *offline* discussion with @psychocoderHPC - please avoid any getter that just do renaming of getter you call `calc...`